### PR TITLE
feat: job handler & space configure plugins

### PR DESCRIPTION
**Pull request by carlbrugger**
This PR introduces two new plugins: `@flatfile/plugin-job-handler` and `@flatfile/plugin-space-configure`.

Docs ref: https://github.com/FlatFilers/Guides/pull/731

### CODE REVIEW

File: plugins/job-handler/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@flatfile/plugin-job-handler",
+  "version": "0.0.0",
+  "description": "A plugin for handling Flatfile Jobs.",
+  "registryMetadata": {
+    "category": "handlers"
```
Comment (carlbrugger) [inline plugins/job-handler/package.json:6]: @ashleygmulligan2 / @bangarang  do you have a preference for a category here?

File: plugins/space-configure/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@flatfile/plugin-space-configure",
+  "version": "0.0.0",
+  "description": "A plugin for configuring a Flatfile Space.",
+  "registryMetadata": {
+    "category": "configure"
```
Comment (carlbrugger) [inline plugins/space-configure/package.json:6]: @ashleygmulligan2 / @bangarang  do you have a preference for a category here?

**Comment by ashleygmulligan2** [comment]
nice, can i do more than create a workbook when space created?

**Comment by ashleygmulligan2** [comment]
how does the code look for success/fail on the job? How do i give you the outcome messages etc?

**Comment by carlbrugger** [comment]
> nice, can i do more than create a workbook when space created?

`configureSpace()` takes a partial `CreateWorkbookConfig` and an optional partial `SpaceConfig`, so it can create sheets, actions, etc. It specifically listens for `space:configure`.

**Comment by carlbrugger** [comment]
> how does the code look for success/fail on the job? How do i give you the outcome messages etc?

Success is assumed if no error is thrown. To fail the job, throw an exception. The callback can return a `Flatfile.JobOutcome` (i.e. `{ info: 'Space configured' }` or `{ outcome: { message: 'Job complete', }, }`)

**Comment by ashleygmulligan2** [comment]
Do you mind making docs PR at same time? That will help me understand the DXP a lil more

File: plugins/job-handler/package.json (hunk 1 above)
Comment (ashleygmulligan2) [reply plugins/job-handler/package.json:6]: let's do core/

File: plugins/space-configure/package.json (hunk 2 above)
Comment (ashleygmulligan2) [reply plugins/space-configure/package.json:6]: let's do core/

**Comment by carlbrugger** [comment]
> Do you mind making docs PR at same time? That will help me understand the DXP a lil more

@ashleygmulligan2 https://github.com/FlatFilers/Guides/pull/731

**Review by ashleygmulligan2 — APPROVED** [review]
live review

**Pull request merged**